tests/lava/magma/core/model/test_py_model.py

```diff
@@ -411,14 +432,73 @@ def test_build_with_dangling_ports(self):
         # Validate that the Process with no OutPorts indeed has no output
         # CspPort
         self.assertIsInstance(
-            pm_with_no_out_ports.in_port._csp_ports[0], FakeCspPort)
-        self.assertEqual(pm_with_no_out_ports.out_port._csp_ports, [])
+            pm_with_no_out_ports.in_port._csp_recv_ports[0], FakeCspPort)
+        self.assertEqual(pm_with_no_out_ports.out_port._csp_send_ports, [])
 
         # Validate that the Process with no InPorts indeed has no input
         # CspPort
-        self.assertEqual(pm_with_no_in_ports.in_port._csp_ports, [])
+        self.assertEqual(pm_with_no_in_ports.in_port._csp_recv_ports, [])
         self.assertIsInstance(
-            pm_with_no_in_ports.out_port._csp_ports[0], FakeCspPort)
+            pm_with_no_in_ports.out_port._csp_send_ports[0], FakeCspPort)
+
+    def test_set_ref_var_ports(self):
+        """Check RefPorts and VarPorts can be set."""
+
+        # Create a new ProcBuilder
+        b = PyProcessBuilder(PyProcModelRefVar, 0)
+
+        # Create Process for which we want to build PyProcModel
+        proc = ProcRefVar()
+
+        # Normally, the Compiler would create PortInitializers from all
+        # ref ports holding only its name and shape
+        ports = list(proc.ref_ports)
+        ref_ports = [PortInitializer(
+            pt.name,
+            pt.shape,
+            getattr(PyProcModelRefVar, pt.name).d_type,
+            pt.__class__.__name__, 32)
+            for pt in ports]
+        # Similarly, the Compiler would create VarPortInitializers from all
+        # var ports holding only its name, shape and var_name
+        ports = list(proc.var_ports)
+        var_ports = [VarPortInitializer(
+            pt.name,
+            pt.shape,
+            pt.var.name,
+            getattr(PyProcModelRefVar, pt.name).d_type,
+            pt.__class__.__name__, 32, PyRefPort.VEC_DENSE)
+            for pt in ports]
+        # Later, the Runtime, would normally create CspPorts that implements
```

Comment:
- Later? Maybe: "Normally, the Runtime would create CspPorts ...
- No comma after "Runtime"
- implements -> implement

Reply:
Done